### PR TITLE
fix build method migration

### PR DESCRIPTION
**Pull request by TimWhiting**
I'm adding in more complex tests based on problems that I'm seeing when trying to debug #840.

Still haven't been able to reproduce that issue, but my plans for resolving the tests I'm adding here might also solve it.

This one might take a day or so though to put together.

### CODE REVIEW

**Comment by TimWhiting** [comment]
This isn't perfect, but it is much better than before and seems to have resolved #840.

Feel free to merge it when you are ready.

File: packages/riverpod_cli/fixtures/unified_syntax/golden/widgets.dart [hunk 1]
```diff
@@ -250,12 +279,12 @@ class HooksConsumerWatch extends StatelessWidget {
   }
 }
 
-class HooksConsumerSimple extends StatelessWidget {
+class HooksConsumerSimple extends ConsumerWidget {
```
Comment (rrousselGit) [inline packages/riverpod_cli/fixtures/unified_syntax/golden/widgets.dart:282]: Why is this converted?

**Comment by rrousselGit** [comment]
Thanks!
Just a quick comment about why one widget is now converted to ConsumerWidget


Doesn't this means in some cases we'd convert widgets that don't use Riverpod to ConsumerWidget?

**Comment by TimWhiting** [comment]
It only occurs in rare cases when a consumer / hook builder are used in a method that is not the widget's build method. I can try to get rid of it, but it becomes very complex. Let me know if you want it fixed, and I can make it happen.

**Comment by rrousselGit** [comment]
That should be fine. Thanks for your work!

**Pull request merged**